spec/forms/developer_search_form_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe DeveloperSearchForm do
+  describe '#search' do
+    before do
+      @developer = create(:developer, email: 'phanthanhhai07t1@gmail.com')
+      @language = create(:language, code: 'en')
+      @programming_language = create(:programming_language, name: 'ruby')
+      @developer_language = create(:developer_language, developer_id: @developer.id, language_id: @language.id)
+      @developer_programming_language = create(:developer_programming_language, developer_id: @developer.id,
```

Comment:
Accessing an intermediate table directly looks wrong to me.
You actually don't need to do so.
This is a bit different example, but can you take a look at this?
```
create_list(:language, evaluator.languages_count, profiles: [profile])
```
http://www.rubydoc.info/gems/factory_bot/file/GETTING_STARTED.md

Reply:
I fixed it